Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsSubComponentModeAngleCone.cpp

```diff
@@ -0,0 +1,432 @@
+        m_yLinearManipulator = AzToolsFramework::LinearManipulator::MakeShared(worldTransform);
+        m_yLinearManipulator->AddEntityComponentIdPair(m_entityComponentIdPair);
+        m_yLinearManipulator->SetAxis(AZ::Vector3::CreateAxisZ());
+
+        m_zLinearManipulator = AzToolsFramework::LinearManipulator::MakeShared(worldTransform);
+        m_zLinearManipulator->AddEntityComponentIdPair(m_entityComponentIdPair);
+        m_zLinearManipulator->SetAxis(AZ::Vector3::CreateAxisY());
+
+        m_yzPlanarManipulator = AzToolsFramework::PlanarManipulator::MakeShared(worldTransform);
+        m_yzPlanarManipulator->AddEntityComponentIdPair(m_entityComponentIdPair);
+        m_yzPlanarManipulator->SetAxes(AZ::Vector3::CreateAxisY(), AZ::Vector3::CreateAxisZ());
+
+        ConfigureLinearView(
+            Internal::ArrowLength, AZ::Color(1.0f, 0.0f, 0.0f, 1.0f), AZ::Color(0.0f, 1.0f, 0.0f, 1.0f), AZ::Color(0.0f, 0.0f, 1.0f, 1.0f));
+
+        ConfigurePlanarView(AZ::Color(0.0f, 1.0f, 0.0f, 1.0f), AZ::Color(0.0f, 0.0f, 1.0f, 1.0f));
+
+        // Position and orientate manipulators
+        AZ::Transform displacementTransform = localTransform;
+        AZ::Vector3 displacementTranslate = localRotation.TransformVector(AZ::Vector3(Internal::ConeHeight, 0.0f, 0.0f));
+        displacementTransform.SetTranslation(localTransform.GetTranslation() + displacementTranslate);
+
+        m_yLinearManipulator->SetLocalTransform(displacementTransform);
+        m_zLinearManipulator->SetLocalTransform(displacementTransform);
+        m_yzPlanarManipulator->SetLocalTransform(displacementTransform);
+
+        // Initialize rotation manipulator for rotating cone
+        m_xRotationManipulator = AzToolsFramework::AngularManipulator::MakeShared(worldTransform);
+        m_xRotationManipulator->AddEntityComponentIdPair(m_entityComponentIdPair);
+        m_xRotationManipulator->SetAxis(AZ::Vector3::CreateAxisX());
+        m_xRotationManipulator->SetLocalTransform(localTransform);
+
+        const AZ::Color xRotationManipulatorColor = AZ::Color(1.0f, 0.0f, 0.0f, 1.0f);
+        m_xRotationManipulator->SetView(AzToolsFramework::CreateManipulatorViewCircle(
+            *m_xRotationManipulator, xRotationManipulatorColor, Internal::XRotationManipulatorRadius, Internal::XRotationManipulatorWidth,
+            AzToolsFramework::DrawHalfDottedCircle));
+
+        AZStd::shared_ptr<JointsComponentModeCommon::SubComponentModes::AngleModesSharedRotationState> sharedRotationState =
+            AZStd::make_shared<JointsComponentModeCommon::SubComponentModes::AngleModesSharedRotationState>();
+
+        struct SharedState
+        {
+            AngleLimitsFloatPair m_startValues;
+        };
+        auto sharedState = AZStd::make_shared<SharedState>();
+
+        m_yLinearManipulator->InstallLeftMouseDownCallback(
+            [this, sharedState](const AzToolsFramework::LinearManipulator::Action& /*action*/) mutable
```

Comment:
nit: This could be updated to `[[maybe_unused]]` which might be useful for debugging in future